src/Collision.cc

```diff
@@ -200,6 +218,73 @@ sdf::SemanticPose Collision::SemanticPose() const
       this->dataPtr->poseRelativeToGraph);
 }
 
+/////////////////////////////////////////////////
+Errors Collision::CalculateInertial(gz::math::Inertiald &_inertial,
+                                    const ParserConfig &_config)
+{
+  Errors errors;
+
+  // Set the density value for the collision material
+  if (this->dataPtr->sdf->HasElement("density"))
+  {
+    this->dataPtr->density = this->dataPtr->sdf->Get<double>("density");
+  }
+  else
+  {
+    // If the density element is missing, let the user know that a default
+    // value would be used according to the policy
+    Error densityMissingErr(
+      ErrorCode::ELEMENT_MISSING,
+      "Collision is missing a <density> child element. "
+      "Using a default density value of 1000.0 kg/m^3. "
+    );
+    enforceConfigurablePolicyCondition(
+      _config.WarningsPolicy(), densityMissingErr, errors
+    );
+  }
+
+  auto geomInertial =
+    this->dataPtr->geom.CalculateInertial(this->dataPtr->density,
+                                          _config, errors);
+
+  if (!geomInertial)
+  {
+    errors.push_back({ErrorCode::LINK_INERTIA_INVALID,
+        "Inertia Calculated for collision: " +
+        this->dataPtr->name + " seems invalid."});
+  }
+  else
+  {
+    _inertial = geomInertial.value();
+
+    // If geometry type is not mesh than calculate inertial pose in Link frame
+    // considering collision frame to be same as inertial frame
+    // In case of mesh the custom inertia calculator should return
+    // the inertial object with the pose already set
+    if (this->dataPtr->geom.Type() != GeometryType::MESH)
```

Comment:
Don't we need to account for `//collision/pose` for meshes as well?

Reply:
Since the origin of the mesh can be set anywhere in the mesh by the user (and that is not known by sdformat), I thought it would be best to leave the calculation of the inertial pose for the custom calculator (where all the mesh vertex data is available). This way the custom calculator would be responsible for setting the pose for the inertial in the link frame. What are your thoughts regarding this?

Reply:
The origin of the mesh would be coinincident with the frame of the collision. For example, if the mesh origin is (0,0,0), but `//collision/pose` is (10, 0, 0), when the mesh is loaded, the final pose of the mesh origin would be 10,0,0. So I would think we need to take that into account somehow since the MOI calculator would not know `//collision/pose`.

Reply:
Ohh okay. Makes sense. I have removed the condition to check for geometry type before resolving the inertial pose in [b05e267](https://github.com/gazebosim/sdformat/pull/1299/commits/b05e267fa77d19cda8f488f904b56e53a797beeb)